pkg/agents/team7/agent1/agent.go

```diff
@@ -79,99 +96,507 @@ func New(baseAgent *infra.Base) (infra.Agent, error) {
+		case targetWeakFood > 0:
+			a.Log("CASE4------------------------------------------------------------------------------------------------------------------------")
+			foodtotake = targetSatisficedFood - food.FoodType(kindnessAdjuster/5) + food.FoodType(greedinessAdjuster)
+
+		// Lowest priority case - it is the standard case, in the absence of messages and treaties, and the agent is not critical
+		default:
+			a.Log("Default CASE------------------------------------------------------------------------------------------------------------------------")
+			foodtotake = targetSatisficedFood - food.FoodType(kindnessAdjuster) + food.FoodType(greedinessAdjuster)
+		}
+
+		// Bound foodtotake
+		if foodtotake > targetFullFood {
+			foodtotake = targetFullFood
+		}
+		a.Log("VWXYZ:", infra.Fields{"food": foodtotake})
+		// Eat Food and Update Statuses
+		Eaten, err := a.TakeFood(foodtotake)
+
+		if err != nil {
+			switch err.(type) {
+			case *infra.FloorError:
+			case *infra.NegFoodError:
+			case *infra.AlreadyEatenError:
+			default:
+			}
+		}
+
+		a.opMem.foodEaten = Eaten
+		if a.opMem.foodEaten > 0 {
+			a.behaviour.greediness -= int(0.5 * float64(a.opMem.daysHungry*a.opMem.daysHungry))
+			a.opMem.daysHungry = 0
+		} else {
+			a.behaviour.greediness -= int(0.3 * float64(a.opMem.daysHungry*a.opMem.daysHungry))
+			a.opMem.daysHungry++
+			a.behaviour.greediness += int(0.3 * float64(a.opMem.daysHungry*a.opMem.daysHungry))
+		}
+
+		a.Log("Team 7 has seen the platform:", infra.Fields{"foodEaten": a.opMem.foodEaten, "health": a.HP(), "daysHungry": a.opMem.daysHungry})
+		a.opMem.seenPlatform = true
+	}
+
+}
+
+func (a *CustomAgent7) manageFloor() {
+	healthInfo := a.HealthInfo()
+	currentFloor := a.Floor()
+	if len(a.opMem.orderPrevFloors) == 0 || a.opMem.orderPrevFloors[len(a.opMem.orderPrevFloors)-1] != currentFloor { //If day 1 or floor change
+
+		a.Log("checking floor change........................................................................................")
+		if a.Age() == 0 {
+			a.Log("Personality Makeup:", infra.Fields{"A": a.personality.agreeableness, "C": a.personality.conscientiousness, "E": a.personality.extraversion, "N": a.personality.neuroticism, "O": a.personality.openness})
+		}
+
+		// Reset Variables on Floor Change
+		a.opMem.currentDayonFloor = 1 // reset currentDay counter
+		a.opMem.recievedReq = false   // reset Requests
+		a.opMem.takeFood = 0
+		a.opMem.leaveFood = 0
+
+		// If the floor tracker is not empty
+		if len(a.opMem.orderPrevFloors) != 0 {
+
+			// ------------------ Run() Block A.1: Estimates the food available on the new floor based on past experience ------------------
+
+			closestFloorAboveCurrent := 0
+			closestFloorBelowCurrent := math.Inf(1)
+			beenOnCurrentFloorBefore := false
+
+			for i := 0; i < len(a.opMem.orderPrevFloors) && !beenOnCurrentFloorBefore; i++ {
+				if a.opMem.orderPrevFloors[i] == currentFloor {
+					beenOnCurrentFloorBefore = true
+				}
+			}
```

Comment:
```suggestion
			for i := 0; i < len(a.opMem.orderPrevFloors) i++ {
				if a.opMem.orderPrevFloors[i] == currentFloor {
					beenOnCurrentFloorBefore = true
					break
				}
			}
```
Wrote the why in other review (but this is where it is after the update)